File: medium/merge_intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> ans;
        int n = intervals.size();

        vector<int> cur = intervals[0];
        for (int i = 1; i < n; i++) {
            if (cur[1] >= intervals[i][0]) {
                cur[1] = max(cur[1], intervals[i][1]);
            }
            else {
                ans.push_back(cur);
                cur = intervals[i];
            }
        }

        ans.push_back(cur);
        return ans;
    }
};

File: medium/merge_intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if not intervals:
            return None
        sortedIntervals = sorted(intervals, key=lambda x: x[0])
        
        result = []
        
        curInterval = sortedIntervals[0]
        i = 1
        while i < len(intervals):
            if sortedIntervals[i][0] <= curInterval[1]:
                curInterval[1] = max(curInterval[1], sortedIntervals[i][1])
            else:
                result.append(curInterval)
                curInterval = sortedIntervals[i]
            i += 1
        result.append(curInterval)
        return result